File: dart/prime-factors/lib/prime_factors.dart
class PrimeFactors {
    List<int> factors(int number) {
        int divisor = 2;
        List<int> r = [];
        while (number > 1) {
            while(number % divisor == 0) {
                r.add(divisor);
                number ~/= divisor;
            }
            ++divisor;
        }
        return r;
    }
}

File: nim/prime-factors/prime_factors.nim
proc primeFactors *(n: int64): seq[int] =
    var dup = n
    var divisor = 2
    var factors: seq[int]
    while dup > 1:
        while dup mod divisor == 0:
            factors.add(divisor)
            dup = int64(dup div divisor)
        inc(divisor)
    return factors
